### PR TITLE
Add test to ensure subgraph edges are correct given an edge direction

### PR DESCRIPTION
**Pull request by mkolodner-sc**
**Scope of work done**

- This test adds an explicit test validating the correctness of the produced edge index tensor in a `HeteroData` object

Where is the documentation for this feature?: N/A

Did you add automated tests or write a test plan?

***Updated Changelog.md?*** NO

***Ready for code review?:*** NO

### CODE REVIEW

File: python/tests/unit/distributed/distributed_neighborloader_test.py [hunk 1]
```diff
@@ -1049,6 +1131,98 @@ def test_cora_supervised_node_classification(self):
...
+
+        # Story -> User
+        # 0  ->   1
+        # 1  ->   2
+        # 2  ->   3
+        # 3  ->   4
+        # 4  ->   0
+
+        user_to_story_edge_index = torch.tensor([[0, 1, 2, 3, 4], [0, 1, 2, 3, 4]])
+        story_to_user_edge_index = torch.tensor([[0, 1, 2, 3, 4], [1, 2, 3, 4, 0]])
```
Comment (kmontemayor2-sc) [inline python/tests/unit/distributed/distributed_neighborloader_test.py:1161]: is it possible for us to create some graphviz/etc for this graph and what is expected/etc? 

Can be hard to visualize just looking at COO :P

Comment (kmontemayor2-sc) [reply python/tests/unit/distributed/distributed_neighborloader_test.py:1161]: https://dreampuf.github.io/GraphvizOnline is a handy tool here.

File: python/tests/unit/distributed/distributed_neighborloader_test.py [hunk 2]
```diff
@@ -549,6 +550,87 @@ def _run_cora_supervised_node_classification(
...
+            global_edge_index = torch.stack(
+                [global_src_edge_index, global_dst_edge_index], dim=0
+            )
+
+            # Then, we can compare the global edge index with the possible edge indices that can exist for this edge type
+            assert (
+                edge_type in possible_edge_indices
+            ), f"User HeteroData contains edge type {edge_type} that is not in the expected graph edge types: {list(possible_edge_indices.keys())}"
+            matches = global_edge_index == possible_edge_indices[edge_type]
+            column_matches = matches.all(dim=0)
```
Comment (kmontemayor2-sc) [inline python/tests/unit/distributed/distributed_neighborloader_test.py:600]: do we use this car ever? Also can we document precisely what we're checking for here?
